Crash when calling the method of an object in the map

### The following program `sample.go` triggers an unexpected result

```go
package main

import "fmt"

type I interface {
	Foo() int
}

type S1 struct {
	i int
}

func (s S1) Foo() int { return s.i }

type S2 struct{}

func (s *S2) Foo() int { return 42 }

func main() {
	var Is = map[string]I{
		"foo": S1{21},
		"bar": &S2{},
	}

	for _, s := range Is {
		fmt.Println(s.Foo())
	}
	fmt.Println(Is["foo"].Foo())
}
```


### Expected result

```console
21
42
21
```


### Got

```console
main.go:25:2: panic
run: reflect: call of reflect.Value.Elem on struct Value
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002eea00, 0x40)
	/usr/local/opt/go/libexec/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc0001c7c18)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/interp/interp.go:504 +0xc8
panic(0x1a43ae0, 0xc00000e8a0)
	/usr/local/opt/go/libexec/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc00036c160, 0xc00048ee00, 0xc0001c79f8)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/interp/run.go:185 +0x253
panic(0x1a43ae0, 0xc00000e8a0)
	/usr/local/opt/go/libexec/src/runtime/panic.go:965 +0x1b9
reflect.Value.Elem(0x1a908e0, 0xc000032f80, 0x99, 0x99, 0xc000089350, 0x98)
	/usr/local/opt/go/libexec/src/reflect/value.go:843 +0x1a5
github.com/traefik/yaegi/interp.rangeMap.func1(0xc00036c160, 0xc00009cf00)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/interp/run.go:2736 +0x18d
github.com/traefik/yaegi/interp.runCfg(0xc00048ee00, 0xc00036c160)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc00000c1e0, 0xc00048de00, 0xc00036c000)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/interp/run.go:122 +0x2b0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc00000c1e0, 0xc0002e8480, 0x175, 0x7ffeefbff306, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/interp/interp.go:596 +0x80c
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc00000c1e0, 0x7ffeefbff306, 0x7, 0xc0002e8300, 0x175, 0x0, 0x0, 0xc0001c7cb0)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/interp/interp.go:418 +0x105
main.runFile(0xc00000c1e0, 0x7ffeefbff306, 0x7, 0x0, 0x1, 0x0)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/cmd/yaegi/run.go:149 +0xb1
main.run(0xc000134010, 0x1, 0x1, 0xc00010c101, 0xc000102480)
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/cmd/yaegi/run.go:112 +0x9c5
main.main()
	/go/pkg/mod/github.com/traefik/yaegi@v0.9.20/cmd/yaegi/yaegi.go:144 +0x415
```


### Yaegi Version

v0.9.20

### Additional Notes

This issue is introduced **after** `v0.9.14`

_No response_

--- a/_test/issue-1189.go
+++ b/_test/issue-1189.go
@@ -1,0 +1,30 @@
+package main
+
+type I interface {
+	Foo() int
+}
+
+type S1 struct {
+	i int
+}
+
+func (s S1) Foo() int { return s.i }
+
+type S2 struct{}
+
+func (s *S2) Foo() int { return 42 }
+
+func main() {
+	Is := map[string]I{
+		"foo": S1{21},
+		"bar": &S2{},
+	}
+	n := 0
+	for _, s := range Is {
+		n += s.Foo()
+	}
+	println(n, Is["foo"].Foo())
+}
+
+// Output:
+// 63 21

--- a/_test/issue-1189.go
+++ b/_test/issue-1189.go
@@ -23,8 +23,9 @@ func main() {
 	for _, s := range Is {
 		n += s.Foo()
 	}
-	println(n, Is["foo"].Foo())
+	bar := "bar"
+	println(n, Is["foo"].Foo(), Is[bar].Foo())
 }
 
 // Output:
-// 63 21
+// 63 21 42

--- a/interp/run.go
+++ b/interp/run.go
@@ -1622,27 +1622,6 @@ func getIndexMap(n *node) {
 				dest(f).Set(z)
 				return fnext
 			}
-		case isInterfaceSrc(n.typ):
-			z = reflect.New(n.child[0].typ.val.frameType()).Elem()
-			n.exec = func(f *frame) bltn {
-				v := value0(f).MapIndex(mi)
-				if !v.IsValid() {
-					dest(f).Set(z)
-					return tnext
-				}
-				e := v.Elem()
-				if len(n.typ.field) == 0 {
-					// e is empty interface, do not wrap
-					dest(f).Set(e)
-					return tnext
-				}
-				if e.Type().AssignableTo(valueInterfaceType) {
-					dest(f).Set(e)
-					return tnext
-				}
-				dest(f).Set(reflect.ValueOf(valueInterface{n, e}))
-				return tnext
-			}
 		default:
 			n.exec = func(f *frame) bltn {
 				if v := value0(f).MapIndex(mi); v.IsValid() {
@@ -1666,20 +1645,6 @@ func getIndexMap(n *node) {
 				}
 				dest(f).Set(z)
 				return fnext
-			}
-		case isInterfaceSrc(n.typ):
-			z = reflect.New(n.child[0].typ.val.frameType()).Elem()
-			n.exec = func(f *frame) bltn {
-				if v := value0(f).MapIndex(value1(f)); v.IsValid() {
-					if e := v.Elem(); e.Type().AssignableTo(valueInterfaceType) {
-						dest(f).Set(e)
-					} else {
-						dest(f).Set(reflect.ValueOf(valueInterface{n, e}))
-					}
-				} else {
-					dest(f).Set(z)
-				}
-				return tnext
 			}
 		default:
 			n.exec = func(f *frame) bltn {
@@ -2729,47 +2694,14 @@ func rangeMap(n *node) {
 	if len(n.child) == 4 {
 		index1 := n.child[1].findex  // map value location in frame
 		value = genValue(n.child[2]) // map
-		if isInterfaceSrc(n.child[1].typ) {
-			if len(n.child[1].typ.field) > 0 {
-				n.exec = func(f *frame) bltn {
-					iter := f.data[index2].Interface().(*reflect.MapIter)
-					if !iter.Next() {
-						return fnext
-					}
-					f.data[index0].Set(iter.Key())
-					if e := iter.Value().Elem(); e.Type().AssignableTo(valueInterfaceType) {
-						f.data[index1].Set(e)
-					} else {
-						f.data[index1].Set(reflect.ValueOf(valueInterface{n, e}))
-					}
-					return tnext
-				}
-			} else {
-				// empty interface, do not wrap
-				n.exec = func(f *frame) bltn {
-					iter := f.data[index2].Interface().(*reflect.MapIter)
-					if !iter.Next() {
-						return fnext
-					}
-					f.data[index0].Set(iter.Key())
-					if iter.Value().Elem().IsValid() {
-						f.data[index1].Set(iter.Value().Elem())
-					} else {
-						f.data[index1].Set(reflect.New(interf).Elem())
-					}
-					return tnext
-				}
+		n.exec = func(f *frame) bltn {
+			iter := f.data[index2].Interface().(*reflect.MapIter)
+			if !iter.Next() {
+				return fnext
 			}
-		} else {
-			n.exec = func(f *frame) bltn {
-				iter := f.data[index2].Interface().(*reflect.MapIter)
-				if !iter.Next() {
-					return fnext
-				}
-				f.data[index0].Set(iter.Key())
-				f.data[index1].Set(iter.Value())
-				return tnext
-			}
+			f.data[index0].Set(iter.Key())
+			f.data[index1].Set(iter.Value())
+			return tnext
 		}
 	} else {
 		value = genValue(n.child[1]) // map

